### PR TITLE
HIP-584: Extract MirrorEvmTxProcessor as a singleton bean

### LINKED ISSUE #6612: HIP-584: Reduce memory footprint by making state asynchronous
### **Motivation**

Ongoing support of `eth_call` and `eth_estimateGas` executions would increase the number of users that start using these new functionalities. Eventually this would increase the number of requests against the Archive Node and performance issues might be faced. This epic would aim to reduce the memory footprint and the number of newly created objects by making the in-memory state agnostic of different requests.

### **Implementation details**

- [x] #6619 
- [x] #6618 
- [ ] #6617 

### **Additional bug fixes**
- [x] #6620

### LINKED ISSUE #6617: HIP-584: Analyze thread and heap dump with jmeter
### Problem

We need to make sure that a transaction execution in the Archive Node uses only a single thread in order to perform refactoring of the in-memory state, using ThreadLocal.

Additionally we should inspect what is the heap dump of the application before and after the refactoring to see how much the memory footprint has been reduced.

Expecting the thread dump would help us analyze how the web3 is handling threads and ThreadLocal instances and whether we have memory leaks or threads that continue work in the background.

### Solution

Setup a `jmeter` against the web3 module and using K6 tests load the system with requests, so that we can analyze its memory and thread management performance.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by IvanKavaldzhiev**
**Description**:

This PR reduces greatly the CPU usage utilized by the web3 module and reduces memory footprint as well.
It extracts additional spring singleton beans and makes `MirrorEvmTxProcessor` a singleton as well. That's the most heavy object used by the web3, so it's important to make it performant.

This PR resolves all comments from #7177 and is its continuation.

**Related issue(s)**:

Fixes #6612
Fixes #6617

**Notes for reviewer**:
Locally these changes introduce docker OOM job to restart the web3 container. This is an issue that is going to be investigated for potential memory issues.

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/store/contract/HederaEvmWorldState.java [hunk 1]
```diff
@@ -120,6 +120,7 @@ public void close() {
         // default no-op
     }
 
+    @Named
     public static class Updater extends AbstractLedgerWorldUpdater<HederaEvmMutableWorldState, Account>
```
Comment (steven-sheehy) [inline hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/store/contract/HederaEvmWorldState.java:124]: Some of the parent classes have state. Don't think we can make singleton

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/store/contract/HederaEvmStackedWorldStateUpdater.java [hunk 2]
```diff
@@ -33,13 +33,15 @@
...
 import java.util.Collections;
 import org.apache.tuweni.bytes.Bytes;
 import org.hyperledger.besu.datatypes.Address;
 import org.hyperledger.besu.datatypes.Wei;
 import org.hyperledger.besu.evm.account.Account;
 import org.hyperledger.besu.evm.account.MutableAccount;
 
+@Named
 @SuppressWarnings("java:S107")
 public class HederaEvmStackedWorldStateUpdater
```
Comment (steven-sheehy) [inline hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/store/contract/HederaEvmStackedWorldStateUpdater.java:46]: Some of the parent classes have state. Don't think we can make singleton

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/contracts/execution/MirrorEvmTxProcessorImpl.java [hunk 3]
```diff
@@ -94,7 +104,9 @@ public HederaEvmTransactionProcessingResult execute(CallServiceParameters params
                 params.getValue(),
                 params.getCallData(),
                 params.isStatic(),
-                aliasManager.resolveForEvm(params.getReceiver()));
+                aliasManager.resolveForEvm(params.getReceiver()),
+                params.getReceiver().equals(Address.ZERO),
+                dynamicProperties.fundingAccountAddress());
```
Comment (steven-sheehy) [inline hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/contracts/execution/MirrorEvmTxProcessorImpl.java:109]: Don't think this needs to be passed via method params. The parent already has EvmProperties so can call it directly.

File: hedera-mirror-web3/src/main/java/com/hedera/node/app/service/evm/contracts/execution/HederaEvmTxProcessor.java [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+import org.hyperledger.besu.datatypes.Wei;
+import org.hyperledger.besu.evm.account.MutableAccount;
+import org.hyperledger.besu.evm.frame.MessageFrame;
+import org.hyperledger.besu.evm.gascalculator.GasCalculator;
+import org.hyperledger.besu.evm.processor.AbstractMessageProcessor;
+import org.hyperledger.besu.evm.processor.ContractCreationProcessor;
+import org.hyperledger.besu.evm.processor.MessageCallProcessor;
+import org.hyperledger.besu.evm.tracing.OperationTracer;
+
+public class HederaEvmTxProcessor {
```
Comment (steven-sheehy) [inline hedera-mirror-web3/src/main/java/com/hedera/node/app/service/evm/contracts/execution/HederaEvmTxProcessor.java:37]: Needs javadoc outlining changes from services

File: hedera-mirror-web3/src/main/java/com/hedera/services/store/contracts/precompile/HTSPrecompiledContract.java [hunk 5]
```diff
@@ -296,42 +305,47 @@ void prepareComputation(Bytes input, final UnaryOperator<byte[]> aliasResolver)
...
                     }
                 }
             }
             case AbiConstants.ABI_ID_APPROVE -> {
-                this.precompile = precompileMapper.lookup(functionId).orElseThrow();
-                this.transactionBody = precompile.body(
-                        input, aliasResolver, new ApproveParams(Address.ZERO, senderAddress, null, true));
+                precompile = precompileMapper.lookup(functionId).orElseThrow();
+                contractCallContext.setPrecompile(precompile);
+                contractCallContext.setTransactionBody(precompile.body(
```
Comment (steven-sheehy) [inline hedera-mirror-web3/src/main/java/com/hedera/services/store/contracts/precompile/HTSPrecompiledContract.java:347]: Should move context updating outside the switch so we don't repeat ourselves like other reviewer recommended.

File: hedera-mirror-web3/src/main/java/com/hedera/services/store/contracts/precompile/impl/AbstractWritePrecompile.java [hunk 6]
```diff
@@ -45,13 +43,8 @@ protected AbstractWritePrecompile(
...
 
     @Override
-    public long getGasRequirement(
-            long blockTimestamp,
-            final TransactionBody.Builder transactionBody,
-            final Store store,
-            final HederaEvmContractAliases mirrorEvmContractAliases) {
-        return pricingUtils.computeGasRequirement(
-                blockTimestamp, this, transactionBody, store, mirrorEvmContractAliases);
+    public long getGasRequirement(long blockTimestamp, final TransactionBody.Builder transactionBody) {
```
Comment (steven-sheehy) [inline hedera-mirror-web3/src/main/java/com/hedera/services/store/contracts/precompile/impl/AbstractWritePrecompile.java:46]: In a future PR, we need to revert adding transactionBody to this method and get it from context.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/store/contract/HederaEvmStackedWorldStateUpdater.java (hunk 2 above)
Comment (IvanKavaldzhiev) [reply hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/store/contract/HederaEvmStackedWorldStateUpdater.java:46]: I think you're right. Then we should declare them as factory beans in the EvmConfiguration.

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/store/contract/HederaEvmWorldState.java (hunk 1 above)
Comment (IvanKavaldzhiev) [reply hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/store/contract/HederaEvmWorldState.java]: Fixed

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/store/contract/HederaEvmStackedWorldStateUpdater.java (hunk 2 above)
Comment (IvanKavaldzhiev) [reply hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/store/contract/HederaEvmStackedWorldStateUpdater.java]: Fixed.

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/contracts/execution/MirrorEvmTxProcessorImpl.java (hunk 3 above)
Comment (IvanKavaldzhiev) [reply hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/contracts/execution/MirrorEvmTxProcessorImpl.java:109]: Fixed

File: hedera-mirror-web3/src/main/java/com/hedera/node/app/service/evm/contracts/execution/HederaEvmTxProcessor.java (hunk 4 above)
Comment (IvanKavaldzhiev) [reply hedera-mirror-web3/src/main/java/com/hedera/node/app/service/evm/contracts/execution/HederaEvmTxProcessor.java:44]: Fixed

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Pull request merged**